cmake/AwsCFlags.cmake

```diff
@@ -16,6 +16,7 @@ include(CheckIncludeFile)
 include(CMakeParseArguments) # needed for CMake v3.4 and lower
 
 option(AWS_ENABLE_LTO "Enables LTO on libraries. Ensure this is set on all consumed targets, or linking will fail" OFF)
+option(LEGACY_COMPILER_SUPPORT "Do not use this unless you're trying to build this library on legacy compilers that you shouldn't be using in the first place" OFF)
```

Comment:
Maybe add a more specific comment here about the actual compiler and version this fixes up: "this enables compilation for <blah blah> but versions prior to <blah blah> remain unsupported."

Reply:
fair enough